presto-hive/src/main/java/io/prestosql/plugin/hive/azure/PrestoAzureConfigurationInitializer.java

```diff
@@ -68,6 +71,19 @@ public PrestoAzureConfigurationInitializer(HiveAzureConfig config)
                     adlClientId.isPresent() && adlCredential.isPresent() && adlRefreshUrl.isPresent(),
                     "If any of ADL client ID, credential, and refresh URL are set, all must be set.");
         }
+
+        this.abfsOAuthClientEndpoint = dropEmpty(config.getAbfsOAuthClientEndpoint());
+        this.abfsOAuthClientId = dropEmpty(config.getAbfsOAuthClientId());
+        this.abfsOAuthClientSecret = dropEmpty(config.getAbfsOAuthClientSecret());
+        if (abfsOAuthClientEndpoint.isPresent() || abfsOAuthClientSecret.isPresent() || abfsOAuthClientId.isPresent()) {
+            checkArgument(
+                    abfsOAuthClientEndpoint.isPresent() && abfsOAuthClientId.isPresent() && abfsOAuthClientSecret.isPresent(),
+                    "If any of ABFS OAuth2 Client endpoint, ID, and secret are set, all must be set.");
+        }
+
+        checkArgument(
```

Comment:
can you add unit test for this validation rules?

Reply:
Done, but I might have overdone it.

Reply:
:) Nice. 

I think there you can still add test to validate for `Multiple ABFS authentication methods configured: access key and OAuth2`. 

Just if you want :)